gptel-curl.el

```diff
@@ -29,48 +29,39 @@
 (require 'gptel)
 
 (eval-when-compile
+  (require 'cl-lib)
   (require 'subr-x))
 (require 'map)
 (require 'json)
 
 (defvar gptel-curl--process-alist nil
   "Alist of active GPTel curl requests.")
 
-(defcustom gptel-proxy-path ""
-  "Path to a proxy to use for gptel interactions.
-Passed to curl via --proxy arg, for example 'proxy.yourorg.com:80'
-Leave it empty if you don't use a proxy."
-  :group 'gptel
-  :type 'string)
-
 (defun gptel-curl--get-args (prompts token)
   "Produce list of arguments for calling Curl.
 
 PROMPTS is the data to send, TOKEN is a unique identifier."
-  (let* ((args
-          (list "--location" "--silent" "--compressed" "--disable"))
-         (url (format "https://%s/v1/chat/completions" gptel-host))
+  (let* ((url (format "https://%s/v1/chat/completions" gptel-host))
          (data (encode-coding-string
                 (json-encode (gptel--request-data prompts))
                 'utf-8))
          (headers
           `(("Content-Type" . "application/json")
             ("Authorization" . ,(concat "Bearer " (gptel--api-key))))))
-    (push (format "-X%s" "POST") args)
-    (push (format "-w(%s . %%{size_header})" token) args)
-    ;; (push (format "--keepalive-time %s" 240) args)
-    (push (format "-m%s" 60) args)
-    (push "-D-" args)
-    (when (not (string-empty-p gptel-proxy-path))
-      (push "--proxy" args)
-      (push (format "%s" gptel-proxy-path) args)
-      (push "--proxy-negotiate" args)
-      (push "--proxy-user" args)
-      (push ":" args))
-    (pcase-dolist (`(,key . ,val) headers)
-      (push (format "-H%s: %s" key val) args))
-    (push (format "-d%s" data) args)
-    (nreverse (cons url args))))
+    (append
+     (list "--location" "--silent" "--compressed" "--disable"
+           (format "-X%s" "POST")
+           (format "-w(%s . %%{size_header})" token)
+           (format "-m%s" 60)
+           "-D-"
+           (format "-d%s" data)
+           url)
```

Comment:
To aid with debugging, it will help to move `url` to the end of the argument list, like in the previous code.

So the curl invocation will look like

```sh
curl --location --silent ... --proxy ... -H%s: %s -H%s: %s url
```

Reply:
You can do this by moving the cl-loop up in the list of lists being appended.

Reply:
makes sense, you mean like below?

```
-           (format "-d%s" data)
-           url)
-     (when (not (string-empty-p gptel-proxy-path))
-       (list "--proxy" gptel-proxy-path
+           (format "-d%s" data))
+     (when (not (string-empty-p gptel-proxy))
+       (list "--proxy" gptel-proxy
              "--proxy-negotiate"
              "--proxy-user" ":"))
      (cl-loop for (key . val) in headers
-              collect (format "-H%s: %s" key val)))))
+              collect (format "-H%s: %s" key val))
+     (list url))))
```

Reply:
Yup this will work :+1: